clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -300,22 +301,22 @@ public void destroy(String name) throws CachePersistenceException {
         configuredConsistency
     );
 
-    if (configuration.isAutoCreate()) {
-      try {
-        this.entity.validateCache(cacheId, clientStoreConfiguration);
-      } catch (IllegalStateException e) {
+    try {
+      if (configuration.isAutoCreate()) {
         try {
-          this.entity.createCache(cacheId, clientStoreConfiguration);
-        } catch (CachePersistenceException ex) {
-          throw new ClusteredStoreCreationException("Error creating server-side cache for " + cacheId, ex);
+          this.entity.validateCache(cacheId, clientStoreConfiguration);
```

Comment:
This is opened to races - we most likely need a loop around here to make sure we exit after having validated configuration always.